bpfd-operator/controllers/bpfd-agent/common.go

```diff
@@ -254,12 +254,6 @@ func reconcileProgram(ctx context.Context,
 	// initialize reconciler state
 	r := rec.getRecCommon()
 
```

Comment:
The only issue I can forsee by moving this down is that now we might pull for EVERY bpfProgram rather than once for all bpfPrograms

Reply:
I don't think it does that.  GetBytecode just builds the interface which get's sent to bpfd, which does the image pulling.  So, I think we'll build the interface each time, but I don't think it changes how many times we actually pull the image.